### PR TITLE
Issue22

### PR DESCRIPTION
**Pull request by phrocker**
Improving python bindings to make python a first class part of this project. Using pybind11 we are able to more concretely make Python a component of sharkbite. will require changes throughout to better support pybind11, but those changes will not break public APIs

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -80,4 +79,10 @@ the API.
             cout << "Key is null" << endl;
     }
 ```
-[accumulo]: https://accumulo.apache.org
+
+
+## Python Lib
+The Python library can be installed by simply typing pip install . into the root source directory.
```
Comment (keith-turner) [inline Readme.md]: Does `pip install .` build and install?  When you run the command does it compile any C or C++ code?

Comment (phrocker) [reply Readme.md]: It does build.

File: pythonexample.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import ctypes
+import os
+import traceback
+import sys
+
+parser = ArgumentParser()
+parser.add_argument("-d", "--dll", dest="dll_file",
+                    help="DLL filename", metavar="FILE")
+parser.add_argument("-i", "--instance", dest="instance",
+                    help="DLL filename")
```
[the remark this answers is not in the document]
Comment (phrocker) [reply pythonexample.py]: you're right. Will add some docs and update this in the next commit. Thanks!

File: Readme.md [hunk 3]
```diff
@@ -80,4 +79,10 @@ the API.
...
+Note that our previous implementation in the python directory is deprecated as of V0.5
```
[the remark this answers is not in the document]
Comment (phrocker) [reply Readme.md]: Yes, sorry. The new python is going to become the norm as it's easier to maintain and has a lot of automated protections. The older one is there to maintain backwards compat. Will make that clear. Thanks!

File: pythonexample.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (phrocker) [reply pythonexample.py]: The python API isn't complete, but will probably get this working, clarify docs, clean things up, then create some more concrete follow ons for things like adding users, using iterators, etc. I also have a plan for python iterators to be supported if you think that would be useful.

File: Readme.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (keith-turner) [reply Readme.md]: The readme updates helped me understand what is happening.  The following is my understanding.

One of the python examples is  an example of calling the C code from another language.  The other is a proper python API.

File: pythonexample.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (keith-turner) [reply pythonexample.py]: When working on the design of iterators for Accumulo I remember looking into Python generators and wishing Java had something like that.  At this point I can't even remember enough of the particulars to know why I was so excited about generators.

**Pull request merged**